build.ps1

```diff
@@ -0,0 +1,929 @@
+    -D dispatch_DIR=$DispatchBuildDir\cmake\modules `
+    -D Foundation_DIR=$FoundationBuildDir\cmake\modules `
+    -G Ninja `
+    -S $SourceCache\indexstore-db `
+    -BuildDefaultTarget
+}
+
+function Build-Syntax($Arch)
+{
+  $SwiftBuildDir = Get-ProjectBuildDir $Arch 1
+
+  Build-CMakeProject `
+    -Arch $Arch `
+    -B $BinaryCache\12 `
+    -D CMAKE_BUILD_TYPE=Release `
+    -D CMAKE_Swift_COMPILER=S:/b/1/bin/swiftc.exe `
+    -D CMAKE_Swift_FLAGS="-resource-dir $SwiftBuildDir\lib\swift -L $SwiftBuildDir\lib\swift\windows" `
+    -D CMAKE_INSTALL_PREFIX=$ToolchainInstallRoot\usr `
+    -D CMAKE_MT=mt `
+    -G Ninja `
+    -S $SourceCache\swift-syntax `
+    -BuildDefaultTarget `
+    -Install
+}
+
+function Build-SourceKitLSP($Arch)
+{
+  $SwiftBuildDir = Get-ProjectBuildDir $Arch 1
+  $DispatchBuildDir = Get-ProjectBuildDir $Arch 2
+  $FoundationBuildDir = Get-ProjectBuildDir $Arch 3
+
+  Build-CMakeProject `
+    -Arch $Arch `
+    -B $BinaryCache\13 `
+    -D CMAKE_BUILD_TYPE=Release `
+    -D CMAKE_C_COMPILER=S:/b/1/bin/clang-cl.exe `
+    -D CMAKE_Swift_COMPILER=S:/b/1/bin/swiftc.exe `
+    -D CMAKE_Swift_FLAGS="-resource-dir $SwiftBuildDir\lib\swift -L $SwiftBuildDir\lib\swift\windows" `
+    -D CMAKE_INSTALL_PREFIX=$ToolchainInstallRoot\usr `
+    -D CMAKE_MT=mt `
+    -D dispatch_DIR=$DispatchBuildDir\cmake\modules `
+    -D Foundation_DIR=$FoundationBuildDir\cmake\modules `
+    -D SwiftSystem_DIR=$BinaryCache\2\cmake\modules `
+    -D TSC_DIR=$BinaryCache\3\cmake\modules `
+    -D LLBuild_DIR=$BinaryCache\4\cmake\modules `
+    -D ArgumentParser_DIR=$BinaryCache\6\cmake\modules `
+    -D SwiftCollections_DIR=$BinaryCache\9\cmake\modules `
+    -D SwiftPM_DIR=$BinaryCache\10\cmake\modules `
+    -D IndexStoreDB_DIR=$BinaryCache\11\cmake\modules `
+    -D SwiftSyntax_DIR=$BinaryCache\12\cmake\modules `
+    -G Ninja `
+    -S $SourceCache\sourcekit-lsp `
+    -BuildDefaultTarget `
+    -Install
+}
+
+#-------------------------------------------------------------------
+
+# Compilers
+Build-Toolchain $HostArch
+Build-LLVM $HostArch
+
+foreach ($Arch in $ArchX64,$ArchX86,$ArchARM64)
+{
+  Build-ZLib $Arch
+  Build-XML2 $Arch
+  Build-CURL $Arch
+  Build-ICU $Arch
+  Build-SwiftRuntime $Arch
+  Build-Dispatch $Arch
+  Build-Foundation $Arch
+  Build-XCTest $Arch
+}
+
+Build-SQLite $HostArch
+Build-SwiftSystem $HostArch
+Build-ToolsSupportCore $HostArch
+Build-LLBuild $HostArch
+Build-Yams $HostArch
+Build-ArgumentParser $HostArch
+Build-Driver $HostArch
+Build-Crypto $HostArch
+Build-Collections $HostArch
+Build-PackageManager $HostArch
+Build-IndexStoreDB $HostArch
+Build-Syntax $HostArch
+Build-SourceKitLSP $HostArch
+
+# Switch to swift-driver
+Copy-Item -Force $BinaryCache\7\bin\swift-driver.exe $ToolchainInstallRoot\usr\bin\swift.exe
+Copy-Item -Force $BinaryCache\7\bin\swift-driver.exe $ToolchainInstallRoot\usr\bin\swiftc.exe
+
+$python = "${Env:ProgramFiles(x86)}\Microsoft Visual Studio\Shared\Python39_64\python.exe"
```

Comment:
It might make sense to prefer the `python` in `Path` and then fallback to this.  This is also now out of support ...

Reply:
Agreed. Also I suspect python only get installed there by older version of VS because mine didn't even after checking the python individual component